src/rooms/rooms.service.ts

```diff
@@ -222,4 +240,89 @@ export class RoomsService {
         throw new Error(error);
       });
   }
+
+  async updateRoomQuestionStack(stackId: string, roomId: number) {
+    const roomResult: Room | undefined = await this.roomRepository.findOne({
+      where: { id: roomId },
+    });
+    if (!roomResult) return [null, 'Room Id is not corrected'];
+    const stackResult: QuestionStack | undefined =
+      await this.questionStackRepository.findOne({
+        where: {
+          id: stackId,
+          status: QuestionStackStatus.ACTIVE,
+        },
+      });
+    if (!stackResult) return [null, 'Stack Id is not corrected'];
+    const currentTime = new Date();
+    if (roomResult.openTime > currentTime)
+      return [null, 'Room is started, cant update any more'];
+    const currentStack = roomResult.questionStack;
+    currentStack.status = QuestionStackStatus.ACTIVE;
```

Comment:
![image](https://github.com/F-Code-Club/rode-be-f-code.tech/assets/124784766/5f17e227-2493-4ef2-a7ae-b5df70a208c4)